sklearn/_build_utils/openmp_helpers.py

```diff
@@ -86,9 +86,10 @@ def check_openmp_support():
             # Link test program
             extra_preargs = os.getenv('LDFLAGS', None)
             if extra_preargs is not None:
-                extra_preargs = extra_preargs.split(" ")
-            else:
-                extra_preargs = []
+                extra_preargs = extra_preargs.strip().split(" ")
+                extra_preargs = [
+                    flag for flag in extra_preargs
+                    if flag.startswith(('-L', '-Wl,-rpath', '-l'))]
```

Comment:
I did not know this multiple choice syntax for startswith!

Reply:
why do we need Wl and rpath? Shouldn't L and l be enough?

Or equivalently why did you remove DYLD_LIBRARY_PATH

Reply:
`-Wl,-rpath` single option :)
`-Wl` tells gcc to pass the following option directly to the linker.

When linking a shared library to a program (e.g. openmp) you need, a compile time, to tell the linker where to find it. Then at runtime you need, at runtime, to tell the dynamic loader where to find it.

the `-L` option solves the former. For the latter there are 2 options `LD_LIBRARY_PATH` or `rpath`. `rpath` allows you to add to the program a path where to look for the shared libraries.

`LD_LIBRARY_PATH` is bad practice because it will make the dynamic loader look there for every python import even when importing modules outside sklearn. And not only python, it will apply to all other programs ! It's bad for 2 reasons. First when `LD_LIBRARY_PATH` is set, the dynamic loader looks there first, then it looks in standard locations. So it makes the loading of all other libraries longer. Then, more rarely, if you happen to have in `LD_LIBRARY_PATH` an other implementation of a library used by an other program, that other program will use this one instead (although sometimes it can be useful if you want to test a different implementation of a library).

Moreover, `LD_LIBRARY_PATH` is also used at compile time. It can be used instead of `-L` for instance. But it's a even worse practice. In the current instructions it's set for both compile time and runtime.

Summary: It's bad, it's confusing,using `LD_LIBRARY_PATH` was a bad choice in the first place, `-rpath` is the appropriate thing to do here :)